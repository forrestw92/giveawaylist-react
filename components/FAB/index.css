.fab {
  display: none;
  flex-flow: row nowrap;
  justify-content: center;
  align-content: center;
  position: fixed;
  right: 20px;
  bottom: 65px;
  height: 50px;
  width: 50px;
  border-radius: 50px;
  z-index: 5;
  box-shadow: 0 2px 3px 0 rgba(0, 0, 0, 0.25);
  background: #a8dadc;
}
@media only screen and (max-width: 560px) {
  .fab {
    display: flex;
    align-items: center;
  }
}
