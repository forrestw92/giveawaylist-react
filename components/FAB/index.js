import React from "react";
import { string, func } from "prop-types";
import stylesheet from "./index.css";

class FAB extends React.Component {
  render() {
    const { ariaLabel, image, className, _onClick } = this.props;
    return (
      <div
        className={stylesheet[className]}
        aria-label={ariaLabel}
        onClick={() => _onClick}
      >
        <img src={image} alt={ariaLabel} />
      </div>
    );
  }
}
FAB.propTypes = {
  _onClick: func.isRequired,
  className: string.isRequired,
  image: string.isRequired,
  ariaLabel: string.isRequired
};

export default FAB;
