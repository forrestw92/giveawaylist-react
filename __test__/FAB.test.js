/* eslint-env jest */

import { render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Router from "next/router";
import FAB from "../components/FAB";
const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe("FAB Component", () => {
  const app = render(
    <FAB
      _onClick={() => {}}
      className={"fab"}
      ariaLabel={"Filter"}
      image={"../../static/icons/filter.svg"}
    />
  );
  it("Has Class", () => {
    expect(app.hasClass("fab")).toBe(true);
  });
  it("Has Aria Label", () => {
    expect(app.attr("aria-label")).toEqual("Filter");
  });
  it("Has Image", () => {
    expect(app.find("img").attr("src")).toEqual(
      "../../static/icons/filter.svg"
    );
  });
  it("Image has Alt", () => {
    expect(app.find("img").attr("alt")).toEqual(
      "Filter"
    );
  });
});

describe("FAB Component Snapshot", () => {
  it("Renders", () => {
    const component = renderer.create(
      <FAB
        _onClick={() => {}}
        className={"fab"}
        ariaLabel={"Filter"}
        image={"../../static/icons/filter.svg"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
