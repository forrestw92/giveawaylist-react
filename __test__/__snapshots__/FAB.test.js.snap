// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB Component Snapshot Renders 1`] = `
<div
  aria-label="Filter"
  className="fab"
  onClick={[Function]}
>
  <img
    alt="Filter"
    src="../../static/icons/filter.svg"
  />
</div>
`;
